packages/storage-plus/src/stack.rs

```diff
@@ -0,0 +1,649 @@
+//! An "Stack" is a storage wrapper that guarantees constant-cost appending to and popping
+//! from a list of items in storage.
+//!
+//! This is achieved by storing each item in a separate storage entry. A special key is reserved
+//! for storing the length of the collection so far.
+use std::any::type_name;
+use std::convert::TryInto;
+use std::marker::PhantomData;
+
+use serde::{de::DeserializeOwned, Serialize};
+
+use cosmwasm_std::{to_vec, StdError, StdResult, Storage};
+
+use crate::helpers::must_deserialize;
+
+const LEN_KEY: &[u8] = b"len";
+
+pub struct Stack<'a, T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    /// prefix of the newly constructed Storage
+    namespace: &'a [u8],
+    /// needed if any suffixes were added to the original namespace.
+    /// therefore it is not necessarily same as the namespace.
+    prefix: Option<Vec<u8>>,
+    length: Option<u32>,
+    item_type: PhantomData<T>,
+}
+
+impl<'a, T: Serialize + DeserializeOwned> Stack<'a, T> {
+    /// constructor
+    pub const fn new(prefix: &'a str) -> Self {
+        Self {
+            namespace: prefix.as_bytes(),
+            prefix: None,
+            length: None,
+            item_type: PhantomData,
+        }
+    }
+    /// This is used to produce a new Stack. This can be used when you want to associate an Stack to each user
+    /// and you still get to define the Stack as a static constant
+    pub fn add_suffix(&self, suffix: &str) -> Self {
+        let prefix = if let Some(prefix) = &self.prefix {
+            [prefix.clone(), suffix.as_bytes().to_vec()].concat()
+        } else {
+            [self.namespace.to_vec(), suffix.as_bytes().to_vec()].concat()
+        };
+        Self {
+            namespace: self.namespace,
+            prefix: Some(prefix),
+            length: None,
+            item_type: self.item_type,
+        }
+    }
+}
+
+impl<'a, T: Serialize + DeserializeOwned> Stack<'a, T> {
+    /// gets the length from storage, and otherwise sets it to 0
+    pub fn get_len(&self, storage: &dyn Storage) -> StdResult<u32> {
```

Comment:
nit: Sticking to std's convention:
```suggestion
    pub fn len(&self, storage: &dyn Storage) -> StdResult<u32> {
```